From 372175a1b99eb21b622805cd4f213bb68ec274f6 Mon Sep 17 00:00:00 2001
From: Jordi Guillaumes Pons <jguillaumes@gmail.com>
Date: Tue, 28 Nov 2017 16:25:45 +0100
Subject: [PATCH 27/62] - dpipe: Fixed error checking for setpgrp(). This
 function returns a PID, or -1 in case of error (it never fails on BSD).

---
 src/dpipe.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/dpipe.c b/src/dpipe.c
index e53ce9904..71395d106 100644
--- a/src/dpipe.c
+++ b/src/dpipe.c
@@ -201,8 +201,9 @@ int main(int argc, char *argv[])
 
 	if (daemonize != 0)
 		daemon(0,0);
-	else if (setpgrp() != 0) {
+	else if (setpgrp() == -1) {
 		fprintf(stderr,"Err: cannot create pgrp\n");
+		perror("setpgrp");
 		exit(1);
 	}
 
-- 
2.35.0

